### PR TITLE
feat: upgrade hds version to 2.8.0

### PR DESCRIPTION
**Pull request by melniiv**
## Description :sparkles:
-Update HDS version
-Fix tests

### Closes :no_good_woman:

**[DEV-XXX](https://helsinkisolutionoffice.atlassian.net/browse/DEV-XXX):**

### CODE REVIEW

File: src/domain/enrolment/enrolmentForm/__tests__/EnrolmentForm.test.tsx [hunk 1]
```diff
@@ -105,7 +105,7 @@ test('renders form and user can fill it and submit and form is saved to local st
   userEvent.click(screen.getByRole('option', { name: /2\. luokka/i }));
 
   // // close dropdown
-  userEvent.click(screen.getByRole('button', { name: /luokka-aste/i }));
+  userEvent.click(screen.getAllByRole('button', { name: /luokka-aste/i })[1]);
```
Comment (nikomakela) [inline src/domain/enrolment/enrolmentForm/__tests__/EnrolmentForm.test.tsx:108]: Strange thing and not fancy, but seems to need it - I tested!

File: src/domain/event/__tests__/enrolmentFormIntegration.test.tsx [hunk 2]
```diff
@@ -295,7 +295,7 @@ const fillForm = async ({
   userEvent.type(enrolmentForm.getByLabelText(/ryhmä/i), '4a');
 
   // select grade from dropdown
-  userEvent.click(enrolmentForm.getByRole('button', { name: /luokka-aste/i }));
+  userEvent.click(screen.getAllByRole('button', { name: /luokka-aste/i })[0]);
```
Comment (nikomakela) [inline src/domain/event/__tests__/enrolmentFormIntegration.test.tsx:298]: Not ideal, but seems to be needed

File: src/domain/event/__tests__/enrolmentFormIntegration.test.tsx [hunk 3]
```diff
@@ -307,7 +307,7 @@ const fillForm = async ({
   userEvent.click(grade2);
 
   // close dropdown
-  userEvent.click(enrolmentForm.getByRole('button', { name: /luokka-aste/i }));
+  userEvent.click(screen.getAllByRole('button', { name: /luokka-aste/i })[0]);
```
Comment (nikomakela) [inline src/domain/event/__tests__/enrolmentFormIntegration.test.tsx:310]: Not ideal, but seems to be needed

File: src/domain/newsletter/__tests__/SubscribeNewsletterPage.test.tsx [hunk 4]
```diff
@@ -109,10 +105,10 @@ describe('SubscribeNewsletterPage', () => {
 
     await waitFor(() => {
       expect(
-        screen.getByRole('heading', {
-          name: /valitettavasti tilauksen lähettäminen epäonnistui! yritäthän myöhemmin uudelleen\./i,
-        })
-      ).toBeInTheDocument();
+        screen.getAllByText(
```
Comment (nikomakela) [inline src/domain/newsletter/__tests__/SubscribeNewsletterPage.test.tsx]: This should be reverted. I tested with the old implementation and it's working ok

File: src/domain/newsletter/__tests__/SubscribeNewsletterPage.test.tsx [hunk 5]
```diff
@@ -73,11 +73,7 @@ describe('SubscribeNewsletterPage', () => {
 
     await waitFor(
       () => {
-        expect(
-          screen.getByRole('heading', {
-            name: /tilaus lähetetty/i,
-          })
-        ).toBeInTheDocument();
+        expect(screen.getAllByText(/tilaus lähetetty/i)).toHaveLength(1);
```
Comment (nikomakela) [inline src/domain/newsletter/__tests__/SubscribeNewsletterPage.test.tsx]: This should be reverted. I tested with the old implementation and it's working ok

**Review by nikomakela — APPROVED** [review]
GG

File: src/domain/newsletter/__tests__/SubscribeNewsletterPage.test.tsx (hunk 4 above)
Comment (nikomakela) [reply src/domain/newsletter/__tests__/SubscribeNewsletterPage.test.tsx]: actually, I tried with `GetByText`

File: src/domain/newsletter/__tests__/SubscribeNewsletterPage.test.tsx (hunk 5 above)
Comment (nikomakela) [reply src/domain/newsletter/__tests__/SubscribeNewsletterPage.test.tsx]: actually, I tried with `GetByText`

Comment (melniiv) [reply src/domain/newsletter/__tests__/SubscribeNewsletterPage.test.tsx]: fixed

**Pull request merged**